File: trunk/JAVA/PokerProtocol/src/bluffinmuffin/protocol/commands2/lobby/career/AuthenticateUserCommand.java
package bluffinmuffin.protocol.commands2.lobby.career;

import java.util.StringTokenizer;

import bluffinmuffin.protocol.commands2.lobby.AbstractLobbyCommand;

public class AuthenticateUserCommand extends AbstractLobbyCommand
{
    @Override
    protected String getCommandName()
    {
        return AuthenticateUserCommand.COMMAND_NAME;
    }
    
    public static String COMMAND_NAME = "lobbyCAREER_AUTHENTICATE_USER";
    
    private final String m_username;
    private final String m_password;
    
    public AuthenticateUserCommand(StringTokenizer argsToken)
    {
        m_username = argsToken.nextToken();
        m_password = argsToken.nextToken();
    }
    
    public AuthenticateUserCommand(String p_Username, String p_Password)
    {
        m_username = p_Username;
        m_password = p_Password;
    }
    
    @Override
    public void encode(StringBuilder sb)
    {
        append(sb, m_username);
        append(sb, m_password);
    }
    
    public String encodeResponse(Boolean success)
    {
        return new AuthenticateUserResponse(this, success).encode();
    }
    
    public String getUsername()
    {
        return m_username;
    }
    
    public String getPassword()
    {
        return m_password;
    }
}

File: trunk/C#/PokerProtocol/Commands/Lobby/Career/AuthenticateUserCommand.cs
using System;
using System.Collections.Generic;
using System.Text;
using EricUtility;
using PokerWorld.Game;
using EricUtility.Networking.Commands;

namespace PokerProtocol.Commands.Lobby.Career
{
    public class AuthenticateUserCommand : AbstractLobbyCommand
    {
        public static string COMMAND_NAME = "lobbyCAREER_AUTHENTICATE_USER";
        
        private readonly string m_Username;
        private readonly string m_Password;

        public string Username
        {
            get { return m_Username; }
        }

        public string Password
        {
            get { return m_Password; }
        }

        public AuthenticateUserCommand(StringTokenizer argsToken)
        {
            m_Username = argsToken.NextToken();
            m_Password = argsToken.NextToken();
        }

        public AuthenticateUserCommand(string p_Username, string p_Password)
        {
            m_Username = p_Username;
            m_Password = p_Password;
        }

        public override void Encode(StringBuilder sb)
        {
            Append(sb, m_Username);
            Append(sb, m_Password);
        }

        public string EncodeResponse(bool yes)
        {
            return new AuthenticateUserResponse(this, yes).Encode();
        }
    }
}
